packages/faucet/src/api/webserver.ts

```diff
@@ -58,21 +59,32 @@ export class Webserver {
           // context.request.body is set by the bodyParser() plugin
           const requestBody = context.request.body;
           const creditBody = RequestParser.parseCreditBody(requestBody);
-
           const { address, denom } = creditBody;
 
           if (!isValidAddress(address, constants.addressPrefix)) {
             throw new HttpError(400, "Address is not in the expected format for this chain.");
           }
 
-          const availableTokens = await faucet.availableTokens();
+          const entry: Date | undefined = this.addressCounter.get(address);
```

Comment:
Do you need to make this type explicit? It should be infered to `Date | undefined`